### PR TITLE
chore: Add spotless plugin

### LINKED ISSUE #1716: Apply spotless plugin
Apply the spotless plugin to improve code quality and refactor code according to java code style guideline.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by naman653**
Fixes #1716 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: Added the spotless plugin according to gradle documentation

### CODE REVIEW

**Comment by naman653** [comment]
@liveHarshit did what needed to be done. Please review now.

**Comment by naman653** [comment]
@iamareebjamal Can you please tell what kind of changes should I do?

**Comment by iamareebjamal** [comment]
This is formatting the project with 2 spaces. Have you seen a Java project with 2 spaces?

**Comment by naman653** [comment]
@iamareebjamal @liveHarshit how can I solve these circleci issues? Please review.

**Comment by liveHarshit** [comment]
Check unit tests 
```> Task :app:testPlayStoreDebugUnitTest

com.eventyay.organizer.core.event.list.EventsViewModelTest > shouldLoadEventsSuccessfully FAILED
    java.lang.ExceptionInInitializerError
        Caused by: java.lang.NullPointerException at EventsViewModelTest.java:46

com.eventyay.organizer.core.event.list.EventsViewModelTest > shouldShowEventError FAILED
    java.lang.NoClassDefFoundError

com.eventyay.organizer.core.event.list.EventsViewModelTest > shouldRefreshEventsSuccessfully FAILED
    java.lang.NoClassDefFoundError

454 tests completed, 3 failed

> Task :app:testPlayStoreDebugUnitTest FAILED

FAILURE: Build failed with an exception.
```

**Comment by ShridharGoel** [comment]
@naman653 I have restarted the build, it should pass.

**Comment by naman653** [comment]
@ShridharGoel what was causing the issue?

File: build.gradle [hunk 1]
```diff
@@ -34,6 +34,6 @@ allprojects {
     }
 }
 
-task clean(type: Delete) {
+task delete(type: Delete) {
```
Comment (liveHarshit) [inline build.gradle]: Why?

**Comment by ShridharGoel** [comment]
> what was causing the issue?

@naman653 Sometimes CircleCI fails for the first time due to some tests but passes when the build is restarted.

@iamareebjamal Any idea why this happens?

File: build.gradle (hunk 1 above)
Comment (naman653) [reply build.gradle]: The spotless plugin had a predefined task named clean which conflict with this one, so I changed the name of the task.

Comment (liveHarshit) [reply build.gradle]: @iamareebjamal Please review

File: app/src/main/java/com/eventyay/organizer/core/auth/login/LoginViewModel.java [hunk 2]
```diff
@@ -76,41 +77,51 @@ private void encryptUserCredentials() {
...
+                                    }
+                                }));
     }
 
     public void setBaseUrl() {
-        String baseUrl = sharedPreferenceModel.getString(Constants.SHARED_PREFS_BASE_URL,
-            BuildConfig.DEFAULT_BASE_URL);
+        String baseUrl =
+                sharedPreferenceModel.getString(
+                        Constants.SHARED_PREFS_BASE_URL, BuildConfig.DEFAULT_BASE_URL);
```
Comment (ShridharGoel) [inline app/src/main/java/com/eventyay/organizer/core/auth/login/LoginViewModel.java]: This formatting looks weird. Doesn't it? @liveHarshit

`sharedPreferenceModel.getString(` could have been in the same line as `String baseUrl =`.

Comment (liveHarshit) [reply app/src/main/java/com/eventyay/organizer/core/auth/login/LoginViewModel.java]: Yes

File: app/src/fdroid/java/com/eventyay/organizer/core/event/create/EventDetailsStepOne.java [hunk 3]
```diff
@@ -47,9 +42,16 @@ public static EventDetailsStepOne newInstance() {
...
-    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        binding = DataBindingUtil.inflate(inflater, R.layout.event_details_step_one, container, false);
-        createEventViewModel = ViewModelProviders.of(getActivity(), viewModelFactory).get(CreateEventViewModel.class);
+    public View onCreateView(
+            @NonNull LayoutInflater inflater,
+            @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        binding =
+                DataBindingUtil.inflate(
+                        inflater, R.layout.event_details_step_one, container, false);
```
Comment (ShridharGoel) [inline app/src/fdroid/java/com/eventyay/organizer/core/event/create/EventDetailsStepOne.java]: @naman653 Similarly here also.

File: app/src/main/java/com/eventyay/organizer/core/attendee/checkin/AttendeeCheckInViewModel.java [hunk 4]
```diff
@@ -37,19 +36,33 @@ public void start(long attendeeId) {
...
+                                                == attendeeliveData.getValue().getId())
+                        .flatMap(
+                                filterAttendee ->
+                                        attendeeRepository.getAttendee(
+                                                attendeeliveData.getValue().getId(), false))
+                        .subscribe(
+                                this.attendeeliveData::setValue,
+                                throwable ->
+                                        error.setValue(
+                                                ErrorUtils.getMessage(throwable).toString())));
```
Comment (ShridharGoel) [inline app/src/main/java/com/eventyay/organizer/core/attendee/checkin/AttendeeCheckInViewModel.java]: Same

File: app/src/main/java/com/eventyay/organizer/core/attendee/checkin/AttendeeCheckInViewModel.java [hunk 5]
```diff
@@ -71,15 +84,19 @@ public void toggleCheckIn() {
...
-                }, throwable -> error.setValue(ErrorUtils.getMessage(throwable).toString())));
+                attendeeRepository
+                        .scheduleToggle(attendeeliveData.getValue())
+                        .subscribe(
+                                () -> {
+                                    // Nothing to do
+                                },
+                                throwable ->
+                                        error.setValue(
+                                                ErrorUtils.getMessage(throwable).toString())));
```
Comment (ShridharGoel) [inline app/src/main/java/com/eventyay/organizer/core/attendee/checkin/AttendeeCheckInViewModel.java]: Same

File: app/src/main/java/com/eventyay/organizer/core/auth/login/LoginViewModel.java (hunk 2 above)
Comment (naman653) [reply app/src/main/java/com/eventyay/organizer/core/auth/login/LoginViewModel.java]: I used the Android Open Source Project filter on googleJavaFormat() in the plugin which is causing such kind of formatting. Should I remove the AOSP filter?

**Comment by naman653** [comment]
The wrong formatting is in the cases when the length of a line is exceeded which is by default is 120 characters per line.

**Comment by ShridharGoel** [comment]
> The wrong formatting is in the cases when the length of a line is exceeded which is by default is 100 characters per line.

`}, throwable -> error.setValue(`

@naman653 This doesn't exceed 100 characters. Should have been in one line instead of three lines.

**Comment by naman653** [comment]
Can you suggest something anything to make this appropriate. I couldn't find anything else.

**Comment by adityastic** [comment]
@iamareebjamal we can close this

**Pull request closed without merging**